src/app/screens/confirmBrc20Transaction/index.tsx

```diff
@@ -0,0 +1,138 @@
+  const navigate = useNavigate();
+  const {
+    recipientAddress,
+    estimateFeesParams,
+    estimatedFees: initEstimatedFees,
+    token,
+  }: ConfirmBrc20TransferState = useLocation().state;
+  const { subscribeToResetUserFlow } = useResetUserFlow();
+  useEffect(() => subscribeToResetUserFlow('/confirm-brc20-tx'), [subscribeToResetUserFlow]);
+
+  const [userInputFeeRate, setUserInputFeeRate] = useState('');
+  const [estimatedFees, setEstimatedFees] = useState(initEstimatedFees);
+  const [isFeeLoading, setIsFeeLoading] = useState(false);
+  const [error, setError] = useState<BRC20ErrorCode | ''>('');
+  const debouncedUserInputFeeRate = useDebounce(userInputFeeRate, 300);
+
+  const { txFee, inscriptionFee, totalFee, btcFee } = getFeeValuesForBrc20OneStepTransfer(
+    estimatedFees.valueBreakdown,
+  );
+
+  useEffect(() => {
+    const runEstimate = async () => {
+      setIsFeeLoading(true);
+      setError('');
+      try {
+        const result = await brc20TransferEstimateFees({
+          ...estimateFeesParams,
+          feeRate: Number(debouncedUserInputFeeRate),
+        });
+        setEstimatedFees(result);
+      } catch (err) {
+        const e = err as Error;
+        if (CoreError.isCoreError(e) && (e.code ?? '') in BRC20ErrorCode) {
+          setError(e.code as BRC20ErrorCode);
+          // TODO add logic in from core hook to show what the fee would have been
+        } else {
+          setError(BRC20ErrorCode.SERVER_ERROR);
+        }
+      }
+      setIsFeeLoading(false);
+    };
+    if (debouncedUserInputFeeRate) {
+      runEstimate();
+    }
+  }, [debouncedUserInputFeeRate, estimateFeesParams]);
+
+  /* callbacks */
+  const handleClickConfirm = () => {
+    if (isLedgerAccount(selectedAccount)) {
+      // TODO ledger
+    }
+    // validate brc20 balance again here
+    if (estimateFeesParams.amount > Number(token.balance)) {
+      setError(t('ERRORS.INSUFFICIENT_BALANCE'));
+      return;
+    }
+    const state: ExecuteBrc20TransferState = {
+      recipientAddress,
+      estimateFeesParams,
+      token,
+    };
+    navigate('/execute-brc20-tx', { state });
+  };
+
+  const handleClickCancel = () => {
+    navigate(-1);
+  };
+
+  const handleClickApplyFee: OnChangeFeeRate = (feeRate) => {
+    if (feeRate) {
+      setUserInputFeeRate(feeRate);
+    }
+  };
+
+  return (
+    <>
+      <AccountHeaderComponent disableMenuOption disableAccountSwitch />
+      <ScrollContainer>
+        <ConfirmBrc20TransactionComponent
+          btcFee={btcFee}
+          currentFeeRate={Number(userInputFeeRate) || estimateFeesParams.feeRate}
+          inscriptionFee={inscriptionFee}
+          onClickApplyFee={handleClickApplyFee}
+          onChangeFee={handleClickApplyFee}
+          onClickCancel={handleClickCancel}
+          onClickConfirm={handleClickConfirm}
+          recipients={[
+            { address: recipientAddress, amountSats: new BigNumber(estimateFeesParams.amount) },
```

Comment:
I think it would be better to create a new Recipient type for BRC-20 and not reuse the btc one. `amountSats` is confusing for a BRC-20 amount.

Reply:
let's address this in another PR, when design is finalised for: https://linear.app/xverseapp/issue/ENG-2785/notice-for-an-non-taproot-address-when-sending-brc-20-tokens

Reply:
Shweet 👍